backend/tests/users/test_schema.py

```diff
@@ -28,3 +31,172 @@ def test_works_when_user_is_logged_in(user, graphql_client):
+                    id
+                    email
+                }
+            }
+        }
+        """
+    )
+
+    assert "errors" in resp
+    assert resp["errors"][0]["message"] == "You don't have the permissions"
+    assert resp["data"] is None
+
+
+def test_query_all_users(admin_user, user_factory, graphql_client):
+    graphql_client.force_login(admin_user)
+
+    users = [user_factory() for i in range(0, 5)]
+    users.append(admin_user)
+
+    resp = graphql_client.query(
+        """
+        {
+            users(offset: 0) {
+                totalCount
+                objects {
+                    id
+                    email
+                }
+            }
+        }
+        """
+    )
+
+    assert "errors" not in resp
+    assert resp["data"]["users"]["totalCount"] == len(users)
+    assert len(resp["data"]["users"]["objects"]) == len(users)
+
+    for test_user in users:
+        assert {
+            'id': str(test_user.id),
+            'email': test_user.email
+        } in resp["data"]["users"]["objects"]
+
+
+def test_get_paginated_users(admin_user, user_factory, graphql_client):
+    graphql_client.force_login(admin_user)
+
+    users = [user_factory() for i in range(0, 5)]
+    users.append(admin_user)
+    users = sorted(users, key=lambda u: u.date_joined, reverse=True)
+
+    resp = graphql_client.query(
+        """
+        {
+            users(offset: 0, limit: 2) {
+                totalCount
+                objects {
+                    id
+                    email
+                }
+            }
+        }
+        """
+    )
+
+    assert "errors" not in resp
+    assert resp["data"]["users"]["totalCount"] == len(users)
+    assert len(resp["data"]["users"]["objects"]) == 2
+
+    assert {
+        'id': str(users[0].id),
+        'email': users[0].email
+    } == resp["data"]["users"]["objects"][0]
+
+    assert {
+        'id': str(users[1].id),
+        'email': users[1].email
+    } == resp["data"]["users"]["objects"][1]
+
+    # request another page
+
+    resp = graphql_client.query(
+        """
+        {
+            users(offset: 2, limit: 2) {
+                totalCount
+                objects {
+                    id
+                    email
+                }
+            }
+        }
+        """
+    )
+
+    assert "errors" not in resp
+    assert resp["data"]["users"]["totalCount"] == len(users)
+    assert len(resp["data"]["users"]["objects"]) == 2
+
+    assert {
+        'id': str(users[2].id),
+        'email': users[2].email
+    } == resp["data"]["users"]["objects"][0]
+
+    assert {
+        'id': str(users[3].id),
+        'email': users[3].email
+    } == resp["data"]["users"]["objects"][1]
+
+
+def test_request_users_with_limit_outside_allowed_value(admin_user, user_factory, graphql_client):
+    graphql_client.force_login(admin_user)
+
+    users = [user_factory() for i in range(0, 100)]
+    users.append(admin_user)
+
+    resp = graphql_client.query(
```

Comment:
can you move this to a function? so you don't have to repeat the query so many times :)